### PR TITLE
Graph functions to add installations. 

### PR DESCRIPTION
**Pull request by saisankargochhayat**
## Related Issues and Dependencies
Add's the methods to add installations to our DB and allow removing them. 

Depends on #1764

### CODE REVIEW

File: thoth/storages/graph/postgres.py [hunk 1]
```diff
@@ -3606,6 +3607,54 @@ def _runtime_environment_conf2models(
 
         return hardware_information, software_environment
 
+    def create_github_app_installation(self, install_data: List[dict]) -> int:
```
Comment (fridex) [inline thoth/storages/graph/postgres.py:3610]: Can the method accept parameters directly (not via a dictionary) so the calling protocol is enforced? The return value can be a boolean that states if the given installation already exists.

The iteration for creating multiple installations will be left up to callee to ensure correct number of installations is created.

File: thoth/storages/graph/postgres.py [hunk 2]
```diff
@@ -3606,6 +3607,54 @@ def _runtime_environment_conf2models(
...
+                    session,
+                    slug=repository.get("full_name"),
+                    repo_name=repository.get("name"),
+                    private=repository.get("private"),
+                    installation_id=repository.get("id"),
+                )
+                count += 1 if not already_exist else 0
+            return count
+
+    def delete_github_app_installation(self, uninstall_data: List[dict]) -> int:
```
Comment (fridex) [inline thoth/storages/graph/postgres.py:3639]: What about introducing a flag "`is_active`" here so we deactivate/activate the installation?

Comment (saisankargochhayat) [reply thoth/storages/graph/postgres.py:3639]: Okay, so the logic would be to keep them in the table after they uninstall and just deactivate them.

File: thoth/storages/graph/postgres.py (hunk 1 above)
Comment (saisankargochhayat) [reply thoth/storages/graph/postgres.py:3610]: Yes, let me change that. Thanks!

File: thoth/storages/graph/postgres.py (hunk 2 above)
Comment (saisankargochhayat) [reply thoth/storages/graph/postgres.py:3639]: Should I alter the uninstall method? or change add a method to add to deactivate?

Comment (fridex) [reply thoth/storages/graph/postgres.py:3639]: Check already existing queries - I think most of them use `update_`.

BTW @pacospace I think we are missing this scenario in the naming convention?

**Comment by saisankargochhayat** [comment]
recheck

File: thoth/storages/graph/postgres.py [hunk 3]
```diff
@@ -3606,6 +3607,58 @@ def _runtime_environment_conf2models(
...
+        Example -
+            "installation_id": "236796147",
+            "repo_name": "advisor",
+            "slug": "thoth-station/advisor",
+            "private": False
+
+        Accepts installation details passed down by Github.
+        :rtype: True, False
+        :raises True: if installation existed and was updated
+        :raises False: if installation was newly added.
```
Comment (fridex) [inline thoth/storages/graph/postgres.py:3622]: Booleans are not raised, but returned.

**Comment by fridex** [comment]
@pacospace PTAL

File: thoth/storages/graph/postgres.py (hunk 2 above)
Comment (pacospace) [reply thoth/storages/graph/postgres.py]: Let's add it if missing thanks :)

File: thoth/storages/graph/postgres.py (hunk 3 above)
Comment (saisankargochhayat) [reply thoth/storages/graph/postgres.py]: Thanks!

File: thoth/storages/graph/postgres.py (hunk 2 above)
Comment (saisankargochhayat) [reply thoth/storages/graph/postgres.py]: @pacospace sorry I didn't get what do we need?

Comment (pacospace) [reply thoth/storages/graph/postgres.py]: https://github.com/thoth-station/storages/blob/master/docs/conventions/README.md conventions for naming of the query, if you think you can reuse what we have or we need a new type add in the PR :) @saisankargochhayat

Comment (saisankargochhayat) [reply thoth/storages/graph/postgres.py]: @pacospace Thanks, I have added it in https://github.com/thoth-station/storages/pull/1768/commits/e646c78aebae9ecdf26db3c2b8a7d520dc6aea60

**Comment by saisankargochhayat** [comment]
recheck

**Comment by saisankargochhayat** [comment]
@pacospace do we need any-more changes in this PR? :smile:

**Comment by saisankargochhayat** [comment]
recheck

File: docs/conventions/README.md [hunk 4]
```diff
@@ -34,3 +34,4 @@ Query Name Template in Thoth
 | 8 | ``<object>_exists()`` Query to check if a specific record with specific attributes exists in the database. | ``bool`` |
 | 9 | ``has_<object><statement>()`` Query to check if a specific record has a specific value of an attribute. | ``bool`` |
 | 10 | ``is_<object><flag/statement>`` Returns boolean value indicating the value of the flag or statement. | ``bool`` |
+| 11 | ``update_<objects>_on_<attribute_filter>()`` Update a particular object's attribute, selecting it based on a unique contstraint. | ``bool`` |
```
Comment (pacospace) [inline docs/conventions/README.md:37]: `constraint` typo

File: thoth/storages/graph/postgres.py [hunk 5]
```diff
@@ -3619,6 +3620,58 @@ def _runtime_environment_conf2models(
...
+                    slug=slug,
+                    repo_name=repo_name,
+                    private=private,
+                    installation_id=installation_id,
+                    is_active=True
+                )
+                return newly_added
+
+    def update_kebechet_github_installations_on_is_active(self, slug: str) -> bool:
+        """We deactivate the app on getting an uninstall event.
```
Comment (pacospace) [inline thoth/storages/graph/postgres.py:3658]: `Deactivate the app on getting an uninstall event.`

**Pull request merged**